account/account.go

```diff
@@ -32,9 +32,9 @@ type AccountInterface interface {
 	TransactionHashInvoke(invokeTxn rpc.InvokeTxnType) (*felt.Felt, error)
 	TransactionHashDeployAccount(tx rpc.DeployAccountType, contractAddress *felt.Felt) (*felt.Felt, error)
 	TransactionHashDeclare(tx rpc.DeclareTxnType) (*felt.Felt, error)
-	SignInvokeTransaction(ctx context.Context, tx *rpc.InvokeTxnV1) error
-	SignDeployAccountTransaction(ctx context.Context, tx *rpc.DeployAccountTxn, precomputeAddress *felt.Felt) error
-	SignDeclareTransaction(ctx context.Context, tx *rpc.DeclareTxnV2) error
+	SignInvokeTransaction(ctx context.Context, tx *rpc.BroadcastInvokev1Txn) error
```

Comment:
Let's change that.
The previous and this new versions set a specific broadcasTx version for both SignInvoke and SignDeclare methods (i.e., 'BroadcastInvokev1Txn' and 'BroadcastDeclareTxnV2'), let's require a general BroadcastInvokeTxnType and BroadcastDeclareTxnType instead.